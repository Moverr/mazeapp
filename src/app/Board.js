import React, { Component } from 'react'; 
import Row from './Row';
import InputField from './InputField';

class Board extends Component {

    constructor(props) {
        super(props);
        this.state = {
            numberOfColumns: 0,
            numberOfRows: 0,
            board: null
        }
    }
    
    renderRow(numberOfRows, numberOfColumns) {
        let x = numberOfRows
        return Array.from({ length: x }, (v, k) => k + 1).map((number) => <Row id={number} columns={numberOfColumns} />);
    }
    initPlay() {
        let result = this.renderRow(this.state.numberOfRows, this.state.numberOfColumns);
        this.setState({ board: result })
        //todo: place the perosno randomly :
        
        

    }

    getMiddleArray(rows,columns){
        if(rows !== undefined && columns !== undefined){
            return ((rows * columns) /2 ) -1;
        }
        return null
    }


    render() {

        const board = this.state.board;
        return (
            <div>
                <h1>Play Game </h1>
                Columns : &nbsp;
                <InputField name={this.state.numberOfColumns} value={this.state.numberOfColumns} callback={(e) => this.setState({ numberOfColumns: e.target.value })} type="number" />

                <br />
                <br />
                Rows : &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
                <InputField name={this.state.numberOfRows} value={this.state.numberOfRows} callback={(e) => this.setState({ numberOfRows: e.target.value })} type="number" />

                <br />
                <button onClick={() => this.initPlay()}>Play</button>
                {board}
            </div>
        );
    }
}

export default Board;