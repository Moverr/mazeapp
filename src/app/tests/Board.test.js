import React from 'react';
import { render } from '@testing-library/react';

import { mount } from 'enzyme';
import Board from '../Board';

test('Component Renders Correctly ', () => {
    const wrapper = mount(<Board />);
    expect(wrapper).toMatchSnapshot();

});
