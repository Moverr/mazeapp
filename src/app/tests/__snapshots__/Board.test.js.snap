// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Renders Correctly  1`] = `ReactWrapper {}`;
